packages/relayer-lib/src/listener/cosmos_sdk.rs

```diff
@@ -0,0 +1,88 @@
+//! This module defines the chain listener for 'ibc-go-eureka'.
+
+use futures::future;
+use tendermint::Hash;
+use tendermint_rpc::{Client, HttpClient};
+
+use anyhow::Result;
+
+use crate::{chain::CosmosSdk, events::EurekaEvent};
+
+use super::ChainListenerService;
+
+/// The `ChainListener` listens for events on the Cosmos SDK chain.
+pub struct ChainListener(HttpClient);
+
+impl ChainListener {
```

Comment:
Is it a listener? It seems to be more a of query service, no?

Reply:
Depends on how you interpret the word. I'd say it is not an event-source. But it does fetch txs, blocks, and parses events. So I feel ok calling it a listener, what do you think?

Reply:
Typically, a listener does not actively fetch information, they listen for events. So, if this was an event _sink_ that would be fed events from the chain, it could be a listener. But this one just fetches by itself, so I don't think I would use the listener term here.